src/cupbearer/scripts/conf/train_classifier_conf.py

```diff
@@ -64,6 +35,22 @@ def __post_init__(self):
         if isinstance(self.model, (MLP, CNN)):
             self.model.output_dim = self.num_classes
 
+        # For datasets that are not necessarily deterministic based only on
+        # arguments, this is where validation sets are set to follow train_data
+        if hasattr(self.train_data, "backdoor"):
+            for name, val_config in self.val_data.items():
+                # WanetBackdoor
+                try:
```

Comment:
Also here, could we just check if the backdoors are WaNet ones with `isinstance`? That feels a bit more robust in case we refactor in the future and e.g. rename `warping_strength` or `control_grid`, because it would give us a runtime error in that case (whereas the current code would just silently not sync the control grids)